control/control.go

```diff
@@ -814,7 +848,7 @@ func (p *pluginControl) FetchMetrics(ns []string, version int) ([]core.Cataloged
 }
 
 func (p *pluginControl) GetMetric(ns []string, ver int) (core.CatalogedMetric, error) {
-	return p.metricCatalog.Get(ns, ver)
+	return p.metricCatalog.GetMetric(ns, ver)
 }
 
 func (p *pluginControl) GetMetricVersions(ns []string) ([]core.CatalogedMetric, error) {
```

Comment:
I suppose that metricCatalog.GetMetricVersions(ns []string) is called only for the cataloged metrics, not for "metrics to collect" defined in task manifest which can contain wildcard treated as a query.
@danielscottt - what do You think about it? The argument 'ns' does not come from task manifest, does it?